tests/integration/defs/verl/test_verl_cases.py

```
@@ -126,20 +138,231 @@ def _run_verl_test(test_path, extra_args=None, timeout=600):
+_STEP_LINE = re.compile(r"step:\d+ - ")
+
+
+def _run_verl_train(extra_args, log_file=None, timeout=1800):
+    """Run ``python3 -m verl.trainer.main_ppo`` in VERL_ROOT with Hydra overrides.
+
+    Stdout/stderr are written to ``log_file`` (default: a fresh mktemp).
+    Asserts the subprocess return code is 0 and returns the log path so the
+    caller can grep per-step metrics out of it.
+    """
+    if log_file is None:
+        log_file = tempfile.mktemp(suffix="-verl-train.log")
```

Ruff (0.15.12)

[error] Use of insecure and deprecated function (`mktemp`)

(S306)